store/iterator.go

```diff
@@ -9,40 +9,50 @@ import (
 	"github.com/iov-one/weave/errors"
 )
 
-///////////////////////////////////////////////////////
-// From Items to Iterator
-
 type btreeIter struct {
-	read      <-chan btree.Item
-	stop      chan<- struct{}
-	onceStop  sync.Once
-	onceRead  sync.Once
```

Comment:
Nice job simplifiying logic to no longer require this flow control (just close one place, always)